src/NativeScript/inspector/CachedResource.mm

```diff
@@ -0,0 +1,91 @@
+#include "CachedResource.h"
+#include "MimeTypeHelper.h"
+
+namespace Inspector {
+WTF::HashMap<WTF::String, Inspector::Protocol::Page::ResourceType> CachedResource::m_mimeTypeMap = {
+    { "text/xml", Inspector::Protocol::Page::ResourceType::Document },
+    { "text/plain", Inspector::Protocol::Page::ResourceType::Document },
+    { "application/xhtml+xml", Inspector::Protocol::Page::ResourceType::Document },
+    { "text/css", Inspector::Protocol::Page::ResourceType::Stylesheet },
+    { "text/javascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "text/ecmascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "application/javascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "application/ecmascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "application/x-javascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "application/json", Inspector::Protocol::Page::ResourceType::Script },
+    { "application/x-json", Inspector::Protocol::Page::ResourceType::Script },
+    { "text/x-javascript", Inspector::Protocol::Page::ResourceType::Script },
+    { "text/x-json", Inspector::Protocol::Page::ResourceType::Script },
+    { "text/typescript", Inspector::Protocol::Page::ResourceType::Script },
+};
+
+CachedResource::CachedResource() {}
+
+CachedResource::CachedResource(WTF::String url)
+    : m_content(WTF::emptyString()) {
+    m_url = url;
+
+    NSString* cachedResourcePath = (NSString*)url;
+    NSURL* cachedResourceUrl = [NSURL URLWithString:cachedResourcePath];
+
+    m_mimeType = WTF::String(NativeScript::mimeTypeByExtension([cachedResourceUrl pathExtension]));
```

Comment:
If you only need `NSURL` for `pathExtension` you can use `-[NSString pathExtension]` directly.